client/src/api/block_builder/input_selection/core/mod.rs

```diff
@@ -390,6 +397,12 @@ impl InputSelection {
 
         self.outputs.extend(storage_deposit_returns);
 
+        if !OUTPUT_COUNT_RANGE.contains(&(self.outputs.len() as u16)) {
+            return Err(Error::Block(iota_types::block::Error::InvalidOutputCount(
+                TryIntoBoundedU16Error::Truncated(self.outputs.len()),
+            )));
```

Comment:
I think I would have a dedicated variant in the ISA error tbh, it's not exactly the same error, trying to select too much vs. trying to build a block with too much. And having this packable stuff here is a bit convoluted.